tests/e2e/bad-login.test.ts

```diff
@@ -0,0 +1,40 @@
+import { BASE_URL, CLIENT_ID, SCOPE, TENANT, TREE, USERNAME } from './config';
+
+// Have TS ignore the `page` Puppeteer object
+declare const page: any;
+
+describe('Test bad login flow', () => {
+  beforeAll(async () => {
+    try {
+      const url = new URL(BASE_URL);
+      url.searchParams.set('clientId', CLIENT_ID!);
+      url.searchParams.set('tenant', TENANT!);
+      url.searchParams.set('tree', TREE!);
+      url.searchParams.set('scope', SCOPE!);
+      url.searchParams.set('un', USERNAME);
+      url.searchParams.set('pw', 'Bad_Password');
```

Comment:
While unlikely someone will use "Bad_Password" in their `.env` file, we could safeguard with something like:
```
url.searchParams.set('pw', `sad_${PASSWORD}_panda`);
```